beacon-chain/blockchain/service.go

```diff
@@ -222,28 +226,46 @@ func (c *ChainService) blockProcessing(processedBlock chan<- *types.Block) {
 		// can be received either from the sync service, the RPC service,
 		// or via p2p.
 		case block := <-c.incomingBlockChan:
```

Comment:
Can you explain this logic in a comment?
What is going on when a block is received? 
I would have expected this to simply call `processBlock(block)` but seems to be doing some other stuff.